integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -415,28 +423,40 @@ private int readInternal(String path, ByteBuffer buf, long size, long offset, Fu
     final int sz = (int) size;
     int nread = 0;
     int rd = 0;
-    Long fd = fi.fh.get();
-    final int flags = fi.flags.get();
+    final long fd = fi.fh.get();
     try {
       FileInStream is = mOpenFileEntries.get(fd);
-      if (is == null && AlluxioFuseOpenUtils.getOpenAction(flags) != OpenAction.READ_WRITE) {
-        LOG.error("Cannot find fd for {} in table", path);
-        return -ErrorCodes.EBADFD();
-      }
       if (is == null) {
-        // Fuse open() file with open flags for reading and writing concurrently.
-        // If the first read(), we open file for reading only.
-        // If opened for write() already, error out.
-        // Alluxio supports read only or write only.
-        CreateFileEntry<FileOutStream> ce = mCreateFileEntries.getFirstByField(ID_INDEX, fd);
-        if (ce != null) {
-          LOG.error(String.format("Cannot open file %s with flags 0x%x "
-              + "for reading and writing concurrently", path, flags));
-          return -ErrorCodes.EIO();
+        // Could be first read in open(READ_WRITE)
+        Object lock = mOpenReadWriteLocks.get(fd);
+        if (lock == null) {
+          // The FileInStream can be created by other
+          // concurrent read() operations
+          is = mOpenFileEntries.get(fd);
```

Comment:
this is to prevent two current read() happens together after open(READ_WRITE).
if lock is null, could be because the other read() successfully create the open file entries.
In filebench internal testing, we do find current read() happening.